lib/matplotlib/axes/_axes.py

```diff
@@ -2935,8 +2937,18 @@ def sign(x):
 
         annotations = []
 
-        for bar, err, dat, lbl in itertools.zip_longest(
-                bars, errs, datavalues, labels
+        if np.iterable(padding):
+            # if padding iterable, check length
+            padding_array = np.asarray(padding)
```

Comment:
```suggestion
            padding = np.asarray(padding)
```
I suggest to just call this `padding` as it may end up an array or list and we don't mind which.  It means we replace the original `padding` but that is OK since we were not going to need the original again.